pkg/skaffold/build/kaniko/sources/sources.go

```diff
@@ -35,7 +35,7 @@ type BuildContextSource interface {
 
 // Retrieve returns the correct build context based on the config
 func Retrieve(cfg *latest.KanikoBuild) (BuildContextSource, error) {
-	if cfg.BuildContext.LocalDir != nil {
+	if cfg.BuildContext == nil || cfg.BuildContext.LocalDir != nil {
```

Comment:
Setting default config should be done in https://github.com/GoogleContainerTools/skaffold/blob/e73775c0304872600bf6e7c643fdbdae715ed5f8/pkg/skaffold/schema/latest/defaults.go#L39 so that `BuildContext` is never nil